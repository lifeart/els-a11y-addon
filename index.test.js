const { onComplete } = require('./index');

function createSimplePath(tagName, attributeName) {
    const astPath = {
        node: {
            type: 'TextNode'
        },
        parent: {
            type: 'AttrNode',
            name: attributeName
        },
        parentPath: {
            parent: {
                tag: tagName
            }
        }
    }
    return astPath;
}

describe('[role] attribute autocomplete', () => {
    it('should autocomplete role for [button] tag', () => {
        expect(onComplete('', {
            results: [],
            type: 'template',
            focusPath: createSimplePath('button', 'role')
        })).toMatchSnapshot();
    })
});

describe('[aria] attribute autocomplete', () => {
    it('should autocomplete [aria-dropeffect]', () => {
        expect(onComplete('', {
            results: [],
            type: 'template',
            focusPath: createSimplePath('ul', 'aria-dropeffect')
        })).toMatchSnapshot();
    })
})