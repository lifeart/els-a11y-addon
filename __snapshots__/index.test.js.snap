// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[aria] attribute autocomplete should autocomplete [aria-dropeffect] 1`] = `
Array [
  Object {
    "label": "copy",
  },
  Object {
    "label": "move",
  },
  Object {
    "label": "link",
  },
  Object {
    "label": "execute",
  },
  Object {
    "label": "popup",
  },
  Object {
    "label": "none",
  },
]
`;

exports[`[role] attribute autocomplete should autocomplete role for [button] tag 1`] = `
Array [
  Object {
    "label": "button",
  },
]
`;
